mantidproject/mantid
# Rawdata explorer widget

**Description of work.**
This PR introduces the Raw Data Explorer widget, designed to browse raw data files in a streamlined way.

The widget appears tiled under the workspace list, and allow the user to select a directory and browse starting from there. Every time they click on a file, it is loaded and displayed in what is supposed to be the most appropriate way. Only one preview can exist at a time, so unused ones are closed as soon as they are not necessary.

By holding the control key, the user can accumulate different files, effectively summing them. Un-accumulating is not supported.

When the total memory used goes over 80% by Mantid count, unused workspaces are deleted until the memory goes down to a more reasonable value, starting by the ones that have not been interacted with for the longest time.

Testing this widget mainly consists in using it. It is supposed to support all the raw data that Mantid can load, or handle gracefully errors. In particular, since it has been developed exclusively on Ubuntu, checking that Qt behaves on other system would be good. 

Attached [RDExp.pdf](https://code.ill.fr/scientific-software/mantill/uploads/e6d2c32799c12a26be0ff059c5a53bf4/RDExp.pdf) is the requirement document, which might be useful for reference of expected behaviors and features.


**To test:**

1. Open workbench.
2. Find the Raw data explorer widget, by default it should be placed behind the `Workspaces` widget.
3. Find some data to visualize: UnitTest/ILL data would have different types of plots from 1D plot, 2D plot, `InstrumentView`, while other instruments from outside ILL would be visualized in the `InstrumentView` widget at the moment.
4. You should  be able to navigate the tree with keyboard and each time you change a file, a new visualization should be displayed (or an informative error dialog).
5. Try to accumulate data: select a number of files while holding the `Ctrl` key. Displayed data should be added in the visualization.

Fixes #29249 . 


Release note and documentation added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
